src/main/java/graphql/schema/GraphQLTypeCodeVisitor.java

```diff
@@ -0,0 +1,51 @@
+package graphql.schema;
+
+import graphql.Internal;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+
+import static graphql.Assert.assertTrue;
+import static graphql.util.TraversalControl.CONTINUE;
+
+/**
+ * This ensure that all fields have data fetchers and that unions and interfaces have type resolvers
+ */
+@Internal
+class GraphQLTypeCodeVisitor extends GraphQLTypeVisitorStub {
+    private final GraphQLCodeRegistry.Builder codeRegistry;
+
+    GraphQLTypeCodeVisitor(GraphQLCodeRegistry.Builder codeRegistry) {
+        this.codeRegistry = codeRegistry;
+    }
+
+    @Override
+    public TraversalControl visitGraphQLFieldDefinition(GraphQLFieldDefinition node, TraverserContext<GraphQLType> context) {
```

Comment:
this whole logic  could be even stronger and the dataFetcher could be removed form the FieldDefinition.

Reply:
Yes - but one breaking change per release.

This released we break the ability to "read" a fetcher from a field.  The next release we remove the ability to "set" a fetcher onto a field.  

I have deprecation in place and hence we can do that in MAJOR version + 2